builtin/mainmenu/tab_multiplayer.lua

```diff
@@ -199,16 +237,120 @@ local function main_button_handler(tabview, fields, name, tabdata)
 		return true
 	end
 
+	if fields.btn_mp_search or fields.key_enter_field == "te_search" then
+		tabdata.fav_selected = 1
+		local input = fields.te_search:lower()
+		tabdata.search_for = fields.te_search
+		
+		if input == "" then 
+			menudata.search_result = nil
+			return true 
+		end
+		
+		if #menudata.favorites < 2 then 
+			return true 
+		end
+		
+		menudata.search_result = {}
+		
+		-- setup the filter and keyword list
+		local filters = {}
+		local keywords = {}
+		for word in input:gmatch("%S+") do
+			local index = word:find(":")
+			if index then
+				local filter = word:sub(1,index-1)
+				local setting = word:sub(index+1,-1)
+
+				if filter == "pvp" then
+					filters.pvp = core.is_yes(setting)
+				elseif filter == "creative" then
+					filters.creative = core.is_yes(setting)
+				elseif filter == "version" then
+					if setting ~= "" then
+						filters.version = setting
+					end
+				elseif filter == "damage" then
+					filters.damage = core.is_yes(setting)
+				else
+					table.insert(keywords, word)
+				end
+				
+			else
+				table.insert(keywords, word)
+			end
+		end
+		
+		if #keywords == 0 then
+			local search_result = {}
+			for i = 1, #menudata.favorites do
+				local server = menudata.favorites[i]
+				if check_filters(filters, server) then
+					table.insert(search_result, server)
+				end
+			end
+			menudata.search_result = search_result
+			if #search_result > 0 then
+				local first_server = search_result[1]
+				core.setting_set("address",     first_server.address)
+				core.setting_set("remote_port", first_server.port)
+			end
+			return true
+		end
+					
+		-- Search the serverlist
+		local search_result = {}
+		for i = 1, #menudata.favorites do
+			local server = menudata.favorites[i]
+			if check_filters(filters, server) then
+				local found = 0
+				
+				for k = 1, #keywords do
+					local keyword = keywords[k]
+					if server.name then
+						local name = server.name:lower()
+						local _, count = name:gsub(keyword, keyword)
+						found = found + count * 2
+					end
+			
+					if server.description then
+						local desc = server.description:lower()
+						local _, count = desc:gsub(keyword, keyword)
+						found = found + count 
+					end
+				end
+				if found > 0 then
+					local points = (#menudata.favorites - i) / 5 + found
+					server.points = points
+					table.insert(search_result, server)
+				end
+			end
+		end
+		if #search_result > 0 then
+			-- Order the results by relevance
+			
```

Comment:
no new line needed here. Comment is also not really needed - the function is table.sort